SwiftWisdomTests/StandardLibrary/Numbers/CountableClosedRange+ExtensionsTests.swift

```diff
@@ -0,0 +1,80 @@
+    // original range as expected.
+    //
+    // I've decided to change this approach because:
+    //
+    // 1. Testing the output of real calls to arc4random_uniform means that the unit test will be non-deterministic, and
+    //    therefore could be flaky. What if we had a bug that only occurred with a single specific value of arc4random?
+    //    Our test would pass most of the time, but occasionally fail.
+    //
+    // 2. We're not really interested in testing the behavior of arc4random in the first place, but rather our logic
+    //    that sits on top of that function.
+    //
+    // One approach would be to seed the random number generator with a hard-coded value, but arc4random is not designed
+    // to be seedable.
+    //
+    // Instead I decided to break out an internal version of ip_random that allows the caller to specify a replacement
+    // for arc4random_uniform. This way we can mock out the function to return controlled values. The obvious downside
+    // is that the test is now somewhat coupled to the implementation, and tests an internal helper instead of the true
+    // public interface.
+
+    func testRandomInRange() {
+        let lowerBound = -100
+        let upperBound = 100
+        let testRange = lowerBound...upperBound
+
+        // We do not want to assume that we know the upper bound argument provided to the random number generator
+        // function. Instead we will make one call and capture the input.
+        var capturedRNGUpperBound: Int? = nil
+        let _ = testRange.ip_random(withUniformRNGFunction: { upperBound in
+            capturedRNGUpperBound = upperBound
+            return 0
+        })
+
+        guard let rngUpperBound = capturedRNGUpperBound, rngUpperBound > 0 else {
+            XCTAssert(false, "Expected that RNG function upper bound \(capturedRNGUpperBound) is non-nil and positive.")
+            return
+        }
+
+        // We will test by sampling the ip_random output for every possible mock RNG output within the upper bound.
+        // We test that the following assumptions hold true:
+        // 1. Each RNG call will be made with the same upper bound value as previously captured. If not, it means that
+        //    ip_random is introducing extra non-determinism on top of the RNG function, which is not its job.
+        // 2. Each output value of ip_random() should fall within the test range.
+        // 3. Every possible value of the test range should be generated at least once, and should be generated the same
+        //    number of times as each other value. Otherwise it means we've introduced a bias.
+        var frequencyBySample = [Int : Int]()
+
+        for mockRNGValue in 0..<rngUpperBound {
+            let sample = testRange.ip_random(withUniformRNGFunction: { upperBound in
+                XCTAssertEqual(upperBound, rngUpperBound, "Expected RNG function to be called with a consistent upper bound each time." )
```

Comment:
I am returning `mockRNGValue` as the output of the mock RNG function. It's a stand-in for a possible value that arc4random would output.